ci: Update GitHub Actions workflows to follow principle of least privilege

kartben opened the pull request:
Default to only giving `contents: read` permission in all workflows and then add additional permissions as needed at the job level.

This shall get us a 10/10 for the Token-Permissions category in the OpenSSF score card :)

Note for reviewers, actions/upload-artifact actions/download-artifact are apparently "special" and don't require extra permissions.

aescolar reviewed (commented):
Thanks @kartben 
+1
No complaints from me

kartben committed:
ci: Update GitHub Actions workflows to follow principle of least privilege

Default to `permissions: read-all` in all workflows and then add
additional permissions as needed at the job level

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

fabiobaltieri commented on .github/workflows/backport_issue_check.yml:
```diff
@@ -29,7 +34,7 @@ jobs:
       - name: Run backport issue checker
         env:
-          GITHUB_TOKEN: ${{ secrets.ZB_GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.ZB_GITHUB_TOKENGITHUB_TOKEN }}
```
typo-ish?

kartben replied:
T_T

fabiobaltieri replied:
you mean T_TT_T

fabiobaltieri commented on .github/workflows/report.txt:
is this meant to be here?

kartben replied:
T_T

fabiobaltieri reviewed (changes requested):
I think there's a couple of unintended changes here

kartben commented:
> I think there's a couple of unintended changes here

Sigh -- thanks! Should be better now

nashif commented on .github/workflows/twister.yaml:
```diff
@@ -15,6 +15,9 @@ on:
 
+permissions:
+  contents: read
+
```
shouldn;t this action have `checks: write` permissions in 'Publish Unit Test Results' section below?

kartben replied:
yes! thanks for paying attention
